### DISCUSSION TITLE
Ubuntu server with support to HEIC images

### DISCUSSION THREAD

**Discussion by josuesantamaria** [ImageMagick/ImageMagick#3911] [Development] [answered]
Hi there,

I have a ubuntu server with a Ruby on rails application in which I'm converting HEIC images to jpgs, I'm using mini-magick gem to make use of ImageMagick features. In my development environment (I'm using MAC), everything works fine, but when I uploaded the code to the ubuntu server in a EC2 Amazon instance, the same code is not working. I read that is a known issue, but I haven't found a way to add that support to ubuntu.

My server is giving me this:
`` ''
identify --version
Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org
Copyright: © 1999-2017 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff wmf x xml zlib
`` ''

What is the way to add the HEIC delegate to that server?

Thanks in advance

**Comment by fmw42** [comment, marked as answer]
Install libheif. See https://github.com/strukturag/libheif And then possibly recompile Imagemagick.  Better answers will need to come from the Imagemagick developers. IM 6.9.7.4 is ancient and may not support this feature.

**Reply by josuesantamaria** [reply]
Thanks for your help. In the end, I ended using this https://github.com/SoftCreatR/imei

It is very easy to run on the ubuntu server and does the job
